server/datastore/mysql/certificate_templates.go

```diff
@@ -200,3 +200,82 @@ func (ds *Datastore) BatchDeleteCertificateTemplates(ctx context.Context, certif
 
 	return nil
 }
+
+func (ds *Datastore) GetHostCertificateTemplates(ctx context.Context, hostUUID string) ([]fleet.HostCertificateTemplate, error) {
+	if hostUUID == "" {
+		return nil, errors.New("hostUUID cannot be empty")
+	}
+
+	stmt := `
+SELECT 
+	ct.name, 
+	hct.status
+FROM host_certificate_templates hct
+	INNER JOIN certificate_templates ct ON ct.id = hct.certificate_template_id 
+WHERE hct.host_uuid = ?`
+
+	var hTemplates []fleet.HostCertificateTemplate
+	if err := sqlx.SelectContext(ctx, ds.reader(ctx), &hTemplates, stmt, hostUUID); err != nil {
+		return nil, err
+	}
+	return hTemplates, nil
+}
+
+func (ds *Datastore) GetMDMProfileSummaryFromHostCertificateTemplates(ctx context.Context, teamID *uint) (*fleet.MDMProfilesSummary, error) {
+	var stmt string
+	var args []interface{}
+
+	if teamID != nil && *teamID > 0 {
+		stmt = `
+SELECT
+	hct.status AS status,
+	COUNT(DISTINCT hct.host_uuid) AS n
+FROM host_certificate_templates hct
+INNER JOIN certificate_templates ct ON hct.certificate_template_id = ct.id
+WHERE ct.team_id = ?
+GROUP BY 1`
+		args = append(args, *teamID)
+	} else {
+		stmt = `
+SELECT
+	hct.status AS status,
+	COUNT(DISTINCT hct.host_uuid) AS n
+FROM host_certificate_templates hct
+GROUP BY 1`
```

Comment:
Nit. Recommend for clarity
```suggestion
GROUP BY hct.status`
```